### PR TITLE
[Refactor] Move `getCloudConfiguration` to `ConnectorMetadata` from `Connector`

### PR DESCRIPTION
**Pull request by dirtysalt**
Fixes #issue

This PR move `getCloudConfiguration` to `ConnectorMetadata` from `Connector`

And to do this:
1. let `HdfsEnvrionment` to hold `CloudConfiguration`
2. and pass `HdfsEnvironment` down to `ConnectorMetadata`
3. in future, I think we can move `properties` to `hdfsEvnrionment`, which will make arch easier to reason about.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java [hunk 1]
```diff
@@ -264,5 +265,8 @@ default void createView(CreateViewStmt stmt) throws DdlException {
     default void alterView(AlterViewStmt stmt) throws DdlException, UserException {
     }
 
+    default CloudConfiguration getCloudConfiguration() {
+        return null;
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:269]: throw exception better? Return null is not expected behavior

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java [hunk 2]
```diff
@@ -409,6 +412,7 @@ public void clear() {
 
     interface BatchWrite {
         void addFile(DataFile file);
+
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:415]: rm blank

File: fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonConnector.java [hunk 3]
```diff
@@ -76,8 +75,8 @@ public PaimonConnector(ConnectorContext context) {
         }
         paimonOptions.setString(WAREHOUSE.key(), warehousePath);
 
-        if (this.cloudConfiguration.getCloudType() == CloudType.AWS && this.cloudConfiguration instanceof AWSCloudConfiguration) {
-            AWSCloudConfiguration awsCloudConfiguration = (AWSCloudConfiguration) this.cloudConfiguration;
+        if (cloudConfiguration.getCloudType() == CloudType.AWS && cloudConfiguration instanceof AWSCloudConfiguration) {
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonConnector.java:78]: `if (cloudConfiguration.getCloudType() == CloudType.AWS)` is enough, that means cloudConfiguration must be AWSCloudConfiguration

File: fe/fe-core/src/main/java/com/starrocks/planner/DeltaLakeScanNode.java [hunk 4]
```diff
@@ -94,7 +94,7 @@ private void setupCloudCredential() {
         }
         CatalogConnector connector = GlobalStateMgr.getCurrentState().getConnectorMgr().getConnector(catalog);
         if (connector != null) {
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/planner/DeltaLakeScanNode.java:96]: actually, connector can't be null, just remove if, you can ask @stephen-shelby for confirmation

File: fe/fe-core/src/main/java/com/starrocks/planner/HdfsScanNode.java [hunk 5]
```diff
@@ -94,7 +94,7 @@ private void setupCloudCredential() {
         }
         CatalogConnector connector = GlobalStateMgr.getCurrentState().getConnectorMgr().getConnector(catalog);
         if (connector != null) {
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/planner/HdfsScanNode.java:96]: ditto

File: fe/fe-core/src/main/java/com/starrocks/planner/HudiScanNode.java [hunk 6]
```diff
@@ -78,7 +78,7 @@ private void setupCloudCredential() {
         }
         CatalogConnector connector = GlobalStateMgr.getCurrentState().getConnectorMgr().getConnector(catalog);
         if (connector != null) {
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/planner/HudiScanNode.java:80]: ditto

File: fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java [hunk 7]
```diff
@@ -119,7 +119,7 @@ private void setupCloudCredential() {
         } else {
             CatalogConnector connector = GlobalStateMgr.getCurrentState().getConnectorMgr().getConnector(catalogName);
             if (connector != null) {
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java:121]: ditto

File: fe/fe-core/src/main/java/com/starrocks/planner/PaimonScanNode.java [hunk 8]
```diff
@@ -86,7 +86,7 @@ private void setupCloudCredential() {
         }
         CatalogConnector connector = GlobalStateMgr.getCurrentState().getConnectorMgr().getConnector(catalog);
         if (connector != null) {
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/planner/PaimonScanNode.java:88]: ditto

**Review by Smith-Cruise — COMMENTED** [review]
LGTM

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java (hunk 2 above)
Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:415]: This is automatically formatted.

File: fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonConnector.java (hunk 3 above)
Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonConnector.java:78]: OK. fixed.

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 40 / 52 (76.92%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/HudiScanNode.java|0|1|00.00%|[81]|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|0|1|00.00%|[89]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|0|1|00.00%|[251]|
|:large_blue_circle:|com/starrocks/connector/delta/DeltaLakeMetadata.java|0|4|00.00%|[35, 36, 71, 81]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[270]|
|:large_blue_circle:|com/starrocks/planner/DeltaLakeScanNode.java|0|1|00.00%|[97]|
|:large_blue_circle:|com/starrocks/connector/delta/DeltaLakeMetadataFactory.java|0|1|00.00%|[50]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|1|2|50.00%|[242]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|1|2|50.00%|[216]|
|:large_blue_circle:|com/starrocks/planner/IcebergTableSink.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonConnector.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergConnector.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadataFactory.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/HdfsEnvironment.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadataFactory.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/CatalogConnectorMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/Connector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HiveTableSink.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java [hunk 9]
```diff
@@ -264,5 +266,8 @@ default void createView(CreateViewStmt stmt) throws DdlException {
     default void alterView(AlterViewStmt stmt) throws DdlException, UserException {
     }
 
+    default CloudConfiguration getCloudConfiguration() {
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:269]: Using non-default implementation function better? This means we must implement it.

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:269]: I think it's more common to throw exception when using default

File: fe/fe-core/src/main/java/com/starrocks/planner/DeltaLakeScanNode.java (hunk 4 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/planner/DeltaLakeScanNode.java:96]: YES, you are right. The connector couldn't be null and has been verified in the analyze phase.

File: fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java (hunk 9 above)
Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:269]: Yes. updated.

File: fe/fe-core/src/main/java/com/starrocks/planner/DeltaLakeScanNode.java (hunk 4 above)
Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/planner/DeltaLakeScanNode.java:96]: fixed.

File: fe/fe-core/src/main/java/com/starrocks/planner/HudiScanNode.java (hunk 6 above)
Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/planner/HudiScanNode.java:80]: fixed.

File: fe/fe-core/src/main/java/com/starrocks/planner/HdfsScanNode.java (hunk 5 above)
Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/planner/HdfsScanNode.java:96]: fixed.

File: fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java (hunk 7 above)
Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java:121]: fixed.

File: fe/fe-core/src/main/java/com/starrocks/planner/PaimonScanNode.java (hunk 8 above)
Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/planner/PaimonScanNode.java:88]: fixed.

File: fe/fe-core/src/main/java/com/starrocks/planner/HdfsScanNode.java [hunk 10]
```diff
@@ -93,9 +93,11 @@ private void setupCloudCredential() {
             return;
         }
         CatalogConnector connector = GlobalStateMgr.getCurrentState().getConnectorMgr().getConnector(catalog);
-        if (connector != null) {
-            cloudConfiguration = connector.getCloudConfiguration();
-        }
+        Preconditions.checkState(connector != null,
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/planner/HdfsScanNode.java:96]: extract a common static function in ScanNode?

Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/planner/HdfsScanNode.java:96]: Dear, let's refactor that in next PR.

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 48 / 79 (60.76%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/HudiScanNode.java|0|5|00.00%|[81, 82, 83, 84, 85]|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|0|7|00.00%|[89, 90, 91, 92, 93, 110, 111]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|0|1|00.00%|[251]|
|:large_blue_circle:|com/starrocks/connector/delta/DeltaLakeMetadata.java|0|4|00.00%|[35, 36, 71, 81]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[270]|
|:large_blue_circle:|com/starrocks/planner/DeltaLakeScanNode.java|0|7|00.00%|[96, 97, 98, 99, 100, 257, 258]|
|:large_blue_circle:|com/starrocks/connector/delta/DeltaLakeMetadataFactory.java|0|1|00.00%|[50]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|1|2|50.00%|[242]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|1|2|50.00%|[216]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|5|7|71.43%|[152, 153]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|5|6|83.33%|[384]|
|:large_blue_circle:|com/starrocks/planner/IcebergTableSink.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonConnector.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergConnector.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadataFactory.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/HdfsEnvironment.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadataFactory.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/CatalogConnectorMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/Connector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HiveTableSink.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 48 / 79 (60.76%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/HudiScanNode.java|0|5|00.00%|[81, 82, 83, 84, 85]|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|0|7|00.00%|[89, 90, 91, 92, 93, 110, 111]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|0|1|00.00%|[251]|
|:large_blue_circle:|com/starrocks/connector/delta/DeltaLakeMetadata.java|0|4|00.00%|[35, 36, 71, 81]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[270]|
|:large_blue_circle:|com/starrocks/planner/DeltaLakeScanNode.java|0|7|00.00%|[96, 97, 98, 99, 100, 257, 258]|
|:large_blue_circle:|com/starrocks/connector/delta/DeltaLakeMetadataFactory.java|0|1|00.00%|[50]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|1|2|50.00%|[242]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|1|2|50.00%|[216]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|5|7|71.43%|[152, 153]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|5|6|83.33%|[384]|
|:large_blue_circle:|com/starrocks/planner/IcebergTableSink.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonConnector.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergConnector.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadataFactory.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/HdfsEnvironment.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadataFactory.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/CatalogConnectorMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/Connector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HiveTableSink.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 48 / 79 (60.76%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/HudiScanNode.java|0|5|00.00%|[81, 82, 83, 84, 85]|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|0|7|00.00%|[89, 90, 91, 92, 93, 110, 111]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|0|1|00.00%|[251]|
|:large_blue_circle:|com/starrocks/connector/delta/DeltaLakeMetadata.java|0|4|00.00%|[35, 36, 71, 81]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[270]|
|:large_blue_circle:|com/starrocks/planner/DeltaLakeScanNode.java|0|7|00.00%|[96, 97, 98, 99, 100, 257, 258]|
|:large_blue_circle:|com/starrocks/connector/delta/DeltaLakeMetadataFactory.java|0|1|00.00%|[50]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|1|2|50.00%|[242]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|1|2|50.00%|[216]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|5|7|71.43%|[152, 153]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|5|6|83.33%|[384]|
|:large_blue_circle:|com/starrocks/planner/IcebergTableSink.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonConnector.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergConnector.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadataFactory.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/HdfsEnvironment.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadataFactory.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/CatalogConnectorMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/Connector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HiveTableSink.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by stephen-shelby — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 63 / 79 (79.75%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|0|1|00.00%|[251]|
|:large_blue_circle:|com/starrocks/connector/delta/DeltaLakeMetadata.java|0|4|00.00%|[35, 36, 71, 81]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[270]|
|:large_blue_circle:|com/starrocks/connector/delta/DeltaLakeMetadataFactory.java|0|1|00.00%|[50]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|1|2|50.00%|[242]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|1|2|50.00%|[216]|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|5|7|71.43%|[110, 111]|
|:large_blue_circle:|com/starrocks/planner/DeltaLakeScanNode.java|5|7|71.43%|[257, 258]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|5|7|71.43%|[152, 153]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|5|6|83.33%|[384]|
|:large_blue_circle:|com/starrocks/planner/IcebergTableSink.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonConnector.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergConnector.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HudiScanNode.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadataFactory.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/HdfsEnvironment.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadataFactory.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/CatalogConnectorMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/Connector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HiveTableSink.java|1|1|100.00%|[]|

**Review by miomiocat — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 65 / 79 (82.28%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|0|1|00.00%|[251]|
|:large_blue_circle:|com/starrocks/connector/delta/DeltaLakeMetadata.java|0|4|00.00%|[35, 36, 71, 81]|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[270]|
|:large_blue_circle:|com/starrocks/connector/delta/DeltaLakeMetadataFactory.java|0|1|00.00%|[50]|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|5|7|71.43%|[110, 111]|
|:large_blue_circle:|com/starrocks/planner/DeltaLakeScanNode.java|5|7|71.43%|[257, 258]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|5|7|71.43%|[152, 153]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|5|6|83.33%|[384]|
|:large_blue_circle:|com/starrocks/planner/IcebergTableSink.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonConnector.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergConnector.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HudiScanNode.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadataFactory.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/HdfsEnvironment.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadataFactory.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/CatalogConnectorMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/Connector.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HiveTableSink.java|1|1|100.00%|[]|

**Review by stephen-shelby — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by dirtysalt** [comment]
@mergify backport branch-3.1

**Pull request merged**